question-answering-using-openai/main.bal

```diff
@@ -0,0 +1,123 @@
+import ballerina/http;
+import ballerina/regex;
+import ballerinax/googleapis.sheets;
+import ballerinax/openai;
+
+configurable string sheetsAccessToken = ?;
+configurable string sheetId = ?;
+configurable string sheetName = ?;
+configurable string openAIToken = ?;
+
+final sheets:Client spreadsheetClient = check new ({auth: {token: sheetsAccessToken}});
+final openai:OpenAIClient openaiClient = check new ({auth: {token: openAIToken}});
+
+// Load the data and compute the embeddings when the service starts
+[map<string>, map<decimal[]>] [documents, docEmbeddings] = check loadData(sheetId, sheetName);
+
+service / on new http:Listener(8080) {
+
+    resource function get answer(string question) returns string?|error {
+
+        string prompt = check constructPrompt(question, documents, docEmbeddings);
+        openai:CreateCompletionRequest prmt = {
+            prompt: prompt,
+            model: "text-davinci-003"
+        };
+        openai:CreateCompletionResponse completionRes = check openaiClient->/completions.post(prmt);
+
+        return completionRes.choices[0].text;
+    }
+}
+
+function countWords(string text) returns int {
+    string[] words = regex:split(text, " ");
+    return words.length();
+}
+
+function cosineSimilarity(decimal[] vector1, decimal[] vector2) returns float {
```

Comment:
These functions should be available via the new `ballerina/math.vector` module. Let's not duplicate them here

Reply:
Ack. Will update when the module is available.